File: frontend/src/types/scan.ts
export interface Scan {
  id: string;
  user_id: string;
  target_url: string;
  scan_mode: "quick" | "full";
  status: "pending" | "crawling" | "scanning" | "completed" | "failed" | "cancelled";
  progress_percent: number;
  pages_found: number;
  pages_scanned: number;
  celery_task_id: string | null;
  config: Record<string, unknown> | null;
  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface ScanCreate {
  target_url: string;
  scan_mode: "quick" | "full";
  config?: Record<string, unknown>;
}

File: backend/app/schemas/scan.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, HttpUrl


class ScanCreate(BaseModel):
    target_url: HttpUrl
    scan_mode: str = "quick"
    config: dict | None = None


class ScanResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    target_url: str
    scan_mode: str
    status: str
    progress_percent: int
    pages_found: int
    pages_scanned: int
    celery_task_id: str | None
    config: dict | None
    started_at: datetime | None
    completed_at: datetime | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class ScanStatus(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    status: str
    progress_percent: int
    pages_found: int
    pages_scanned: int
    error_message: str | None
